pints/_log_priors.py

```diff
@@ -165,7 +166,168 @@ def sample(self, n=1):
+    """
+    def __init__(self, a, b):
+        # Parse input arguments
+        self._a = float(a)
+        self._b = float(b)
+
+        # Validate inputs
+        if self._a <= 0:
+            raise ValueError('Shape parameter alpha must be positive')
+        if self._b <= 0:
+            raise ValueError('Shape parameter beta must be positive')
+
+        # Cache constant
+        self._log_beta = scipy.special.betaln(self._a, self._b)
+
+    def __call__(self, x):
+        if x[0] < 0.0 or x[0] > 1.0:
+            return -float('inf')
+        else:
+            return scipy.special.xlogy(self._a - 1.0,
+                                       x[0]) + scipy.special.xlog1py(
+                self._b - 1.0, -x[0]) - self._log_beta
+
+    def evaluateS1(self, x):
+        """ See :meth:`LogPDF.evaluateS1()`. """
+        value = self(x)
+        _x = x[0]
+
+        # Account for pathological edges
+        if _x == 0.0:
+            _x = np.nextafter(0.0, 1.0)
+        elif _x == 1.0:
+            _x = np.nextafter(1.0, 0.0)
+
+        if _x < 0.0 or _x > 1.0:
+            return value, np.asarray([0.])
+        else:
+            # Use np.divide here to better handle possible v small denominators
+            return value, np.asarray([np.divide(self._a - 1., _x) - np.divide(
+                self._b - 1., 1. - _x)])
+
+    def n_parameters(self):
+        """ See :meth:`LogPrior.n_parameters()`. """
+        return 1
+
+    def sample(self, n=1):
+        """ See :meth:`LogPrior.sample()`. """
+        return np.random.beta(self._a, self._b, size=(n, 1))
+
+
+class ExponentialLogPrior(pints.LogPrior):
+    """
+    Defines an exponential (log) prior with given rate parameter ``rate`` with
+    pdf f(x|rate) = rate * e^-{rate*x}.
+
+    For example: ``p = ExponentialLogPrior(0.5, 1)`` for a rate ``rate=0.5``.
+
+    *Extends:* :class:`LogPrior`
+    """
+    def __init__(self, rate):
+        # Parse input arguments
+        self._rate = float(rate)
+
+        # Validate inputs
+        if self._rate <= 0:
+            raise ValueError('Rate parameter "scale" must be positive')
+
+        # Cache constant
+        self._log_scale = np.log(self._rate)
+
+    def __call__(self, x):
+        if x[0] < 0.0:
+            return -float('inf')
+        else:
+            return self._log_scale - self._rate * x[0]
+
+    def evaluateS1(self, x):
+        """ See :meth:`LogPDF.evaluateS1()`. """
+        value = self(x)
+
+        if x[0] < 0.0:
+            return value, np.asarray([0.])
+        else:
+            return value, np.asarray([-self._rate])
+
+    def n_parameters(self):
+        """ See :meth:`LogPrior.n_parameters()`. """
+        return 1
+
+    def sample(self, n=1):
+        """ See :meth:`LogPrior.sample()`. """
+        return np.random.exponential(scale=1. / self._rate, size=(n, 1))
+
+
+class GammaLogPrior(pints.LogPrior):
+    """
+    Defines a gamma (log) prior with given shape parameter ``a`` and rate
```

Comment:
A couple of things: can we make maths for pdf renderable in readthedocs, and can we also add a short note to indicate that this parameterisation is such that E(x|a,b) = a / b. I wish other languages would do this as it makes it quick to determine which parameterisation is being used.